tests/new_upgrades/conftest.py

```diff
@@ -60,10 +61,30 @@ def shared_checkout(shared_name):
     ) as sat_checkout:
         sat_checkout.ready()
         sat_instance = bx_inst.from_inventory(
-            filter=f'@inv._broker_args.upgrade_group == "{shared_name}_shared_checkout"'
-        )[0]
-        sat_instance.setup()
-    return sat_instance
+            filter=f'@inv._broker_args.upgrade_group == "{shared_name}_shared_checkout" |'
+            '@inv._broker_args.workflow == "deploy-satellite"'
+        )
+    return sat_instance[0]
+
+
+def shared_cap_checkout(shared_name):
+    cap_inst = Broker(
+        workflow=settings.CAPSULE.deploy_workflows.product,
+        deploy_sat_version=settings.UPGRADE.FROM_VERSION,
+        host_class=Capsule,
+        upgrade_group=f'{shared_name}_shared_checkout',
+    )
+    with SharedResource(
+        resource_name=f'{shared_name}_cap_checkout',
+        action=cap_inst.checkout,
+        action_validator=lambda result: isinstance(result, Capsule),
+    ) as cap_checkout:
+        cap_checkout.ready()
+        cap_instance = cap_inst.from_inventory(
+            filter=f'@inv._broker_args.upgrade_group == "{shared_name}_shared_checkout" |'
+            '@inv._broker_args.workflow == "deploy-capsule"'
```

Comment:
ditto just `deploy-capsule` should be deploy-upgrade-capsule`

Reply:
Addressed in previous comment.